src/groupeddataframe/splitapplycombine.jl

```diff
@@ -1166,21 +1151,23 @@ julia> by(df, :a, (:b, :c) => x -> (minb = minimum(x.b), sumc = sum(x.c)))
 │ 4   │ 4     │ 1     │ 12    │
 ```
 """
-by(d::AbstractDataFrame, cols::Any, f::Any;
-   sort::Bool=false, skipmissing::Bool=false) =
-    combine(f, groupby(d, cols, sort=sort, skipmissing=skipmissing))
-by(f::Any, d::AbstractDataFrame, cols::Any;
-   sort::Bool=false, skipmissing::Bool=false) =
-    by(d, cols, f, sort=sort, skipmissing=skipmissing)
-by(d::AbstractDataFrame, cols::Any, f::Pair;
-   sort::Bool=false, skipmissing::Bool=false) =
-    combine(f, groupby(d, cols, sort=sort, skipmissing=skipmissing))
+by(d::AbstractDataFrame, cols::Any, f::Function;
+   sort::Bool=false, skipmissing::Bool=false, keepkeys::Bool=true) =
+    combine(f, groupby(d, cols, sort=sort, skipmissing=skipmissing),
+            keepkeys=keepkeys)
+
+by(f::Function, d::AbstractDataFrame, cols::Any;
+   sort::Bool=false, skipmissing::Bool=false, keepkeys::Bool=true) =
+    by(d, cols, f, sort=sort, skipmissing=skipmissing, keepkeys=keepkeys)
+
 by(d::AbstractDataFrame, cols::Any, f::Pair...;
-   sort::Bool=false, skipmissing::Bool=false) =
-    combine(f, groupby(d, cols, sort=sort, skipmissing=skipmissing))
+   sort::Bool=false, skipmissing::Bool=false, keepkeys::Bool=true) =
+    combine(groupby(d, cols, sort=sort, skipmissing=skipmissing),
+            f..., keepkeys=keepkeys)
+
 by(d::AbstractDataFrame, cols::Any;
    sort::Bool=false, skipmissing::Bool=false, f...) =
-    combine(values(f), groupby(d, cols, sort=sort, skipmissing=skipmissing))
+    combine(groupby(d, cols, sort=sort, skipmissing=skipmissing); f...)
 
 """
     aggregate(df::AbstractDataFrame, fs)
```

Comment:
Actually now I have implemented the following extension to work:
```
combine(gd, [:a,:b] .=> mean, [:a, :b] .=> var)
```
(it does not work on 0.20.2 but after this PR it will)

so all we need to do is to add broadcasting for `All` and friends that @nalimilan is working on (it can be post 1.0 release).